src/coreclr/jit/codegencommon.cpp

```diff
@@ -6302,7 +6302,8 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
             }
 
             // Note we are always reading from the original frame here
-            const var_types lclTyp  = genActualType(varDsc->lvType);
+            //
+            const var_types lclTyp  = varDsc->GetActualRegisterType();
```

Comment:
No, it won't be enough for arm64,. There's a lot of work to do to bring up OSR for arm64.

Reply:
I still don't quite understand how this handles multi-reg args on SysV. It seems we will end up generating an illegal load below that tries to use `movaps` between GP regs and memory. Do I misunderstand something?

Reply:
I'm not sure.

For larger SIMD types,  `ins_Load` will return `movups` or `movupd` since we're not claiming the data is aligned.

Reply:
I mean for the following case:
```csharp
struct Foo { public long A, B; }
public static void Bar(Foo f) // f is passed in rdi and rsi on SysV
{
  ...
}
```

What does this code generate in this case?

Reply:
This code is setting the initial values for enregistered args/locals in the OSR frame. Struct promotion is currently disabled for OSR methods so anything that would be a multi-arg reg is not enregistered in the OSR method and is read/written from the original frame slot.

So for code like

```C#
public struct S
{
    public long x;
    public long y;
}

static int Foo(S s)
{
    int result = 0;
    for (int i = 0; i < 100; i++)
    {
        result += (int) (s.x + s.y);
    }
    return result / 100;
}
```
the OSR method is
```asm
; Assembly listing for method X:Foo(S):int
; Emitting BLENDED_CODE for X64 CPU with AVX - Unix
; Tier-1 compilation
; OSR variant for entry point 0x6
; optimized code
; optimized using profile data
; rbp based frame
; fully interruptible
; with PGO: edge weights are valid, and fgCalledCount is 1
; Final local variable assignments
;
;  V00 arg0         [V00,T00] (  4,  4   )  struct (16) [rbp+20H]   do-not-enreg[SFA] multireg-arg single-def
;  V01 loc0         [V01,T02] (  3,  2   )     int  ->  rdx        
;  V02 loc1         [V02,T01] (  3,  3   )     int  ->  rax        
;# V03 OutArgs      [V03    ] (  1,  1   )  lclBlk ( 0) [rsp+00H]   "OutgoingArgSpace"
;  V04 rat0         [V04,T03] (  3,  0   )     int  ->  rdx         "ReplaceWithLclVar is creating a new local variable"
;
; Lcl frame size = 0

G_M5381_IG01:              ;; offset=0000H
       55                   push     rbp
       488BEC               mov      rbp, rsp
       8B551C               mov      edx, dword ptr [rbp+1CH]    // result on original frame (enregistered in OSR)
       8B4518               mov      eax, dword ptr [rbp+18H]    // i on original frame (enregistered in OSR)

G_M5381_IG02:              ;; offset=000AH
       035520               add      edx, dword ptr [rbp+20H]    // s.x on original frame
       035528               add      edx, dword ptr [rbp+28H]    // s.y on original frame
       FFC0                 inc      eax
       83F864               cmp      eax, 100
       7CF3                 jl       SHORT G_M5381_IG02

G_M5381_IG03:              ;; offset=0017H
       BF1F85EB51           mov      edi, 0x51EB851F
       8BC7                 mov      eax, edi
       F7EA                 imul     edx:eax, edx
       8BC2                 mov      eax, edx
       C1E81F               shr      eax, 31
       C1FA05               sar      edx, 5
       03C2                 add      eax, edx
ore 0.00
G_M5381_IG04:              ;; offset=002AH
       5D                   pop      rbp
       4883C428             add      rsp, 40
       5D                   pop      rbp
       C3                   ret      
```
I'll add an assert since somewhere down the road we will want to try and enable struct promotion and field enregistration for OSR methods, and we'll need to make changes here.

Reply:
Ah ok, I see. Thanks.